### ISSUE TITLE
[UnnecessaryEventHandlerParameter] False positive when there is a shadow variable

### ISSUE THREAD

**Issue by BraisGabin** [appKODE/detekt-rules-compose#41]
This component have little sense (infinite recursion) but it's a really easy reproducer of this false positive:

```kotlin
@Composable
fun Asdf(rotation: Float, onClick: (Float) -> Unit) {
    Column {
        Asdf(rotation, { rotation -> onClick(rotation) })
    }
}
```

This code fires `UnnecessaryEventHandlerParameter` but if I change the code to this:

```kotlin
@Composable
fun Asdf(rotation: Float, onClick: (Float) -> Unit) {
    Column {
        Asdf(rotation, { foo -> onClick(foo) })
    }
}
```

The rule works perfectly.

Probably using type solving this could be solvable. The other issue is on detekt itself. The shadowing rule doesn't catch that type of shadowing.

**Comment by dimsuz** [comment]
Interesting! Thanks, I'll investigate this more closely and will think about how this could be solved, maybe it'll require to switch this rule to use type resolution indeed...
